pkg/sources/errors_test.go

```diff
@@ -0,0 +1,149 @@
+package sources
+
+import (
+	"reflect"
+	"sync"
+	"testing"
+)
+
+func TestNewScanErrors(t *testing.T) {
+	testCases := []struct {
+		name     string
+		projects int
+		want     *ScanErrors
+	}{
+		{
+			name:     "no projects",
+			projects: 0,
+			want: &ScanErrors{
+				errors: make([]error, 0),
+			},
+		},
+		{
+			name:     "one project",
+			projects: 1,
+			want: &ScanErrors{
+				errors: make([]error, 0, 1),
+			},
+		},
+		{
+			name:     "fifty projects",
+			projects: 50,
+			want: &ScanErrors{
+				errors: make([]error, 0, 50),
+			},
```

Comment:
Hm, I wonder if these tests don't have enough leniency? For example, does it matter if the underlying slice has exactly 50 allocated here?

I'm wondering too if preallocating the slice is being too pessimistic about scanning. Most of the time there shouldn't be any errors (hopefully).